dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/users/SamlSessionRegistry.java

```diff
@@ -0,0 +1,83 @@
+package edu.harvard.iq.dataverse.users;
+
+import edu.harvard.iq.dataverse.DataverseSession;
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUser;
+import edu.harvard.iq.dataverse.persistence.user.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ejb.Lock;
+import javax.ejb.LockType;
+import javax.ejb.Singleton;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Singleton
+public class SamlSessionRegistry {
+    private static final Logger logger = LoggerFactory.getLogger(SamlSessionRegistry.class);
+
+    private Map<AuthenticatedUser, List<DataverseSession>> userToSessionIndex = new HashMap<>();
```

Comment:
It is possible that this map will grow over time. Since sessions can expire over time without explicit click on "logout" either inside application and in idp. But for now I think it is ok. But still probably some separate task for resolving this issue will still be nedded. What do you think?